### ISSUE TITLE
example script multivariate.py throws "UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape.  This may consume a large amount of memory."

### ISSUE THREAD

**Issue by goughjt** [tensorflow/tensorflow#16626]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
I am using the example script `tensorflow/contrib/timeseries/examples/multivariate.py`
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Mac OS High Sierra (darwin)
- **TensorFlow installed from (source or binary)**:
source
- **TensorFlow version (use command below)**:
v1.5.0-rc1-1781-g86c10063c8 1.5.0-rc1
- **Python version**: 
3.6.4
- **Bazel version (if compiling from source)**:
0.9.0-homebrew
- **GCC/Compiler version (if compiling from source)**:
Apple LLVM version 9.0.0 (clang-900.0.39.2)
- **CUDA/cuDNN version**:
n/a compiled without CUDA support
- **GPU model and memory**:
n/a GPU not supported on Mac with SIP
- **Exact command to reproduce**:
$ python $GOPATH/src/github.com/tensorflow/tensorflow/tensorflow/contrib/timeseries/examples/multivariate.py

### Describe the problem
The script throws warning:
`UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.`

### Source code / logs
Here is the traceback of the warning on my system (I have tensorflow installed in a virtual environment called "tf3".:
```
  File "multivariate.py", line 59, in multivariate_train_and_sample
    estimator.train(input_fn=train_input_fn, steps=training_steps)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/estimator/estimator.py", line 352, in train
    loss = self._train_model(input_fn, hooks, saving_listeners)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/estimator/estimator.py", line 809, in _train_model
    features, labels, model_fn_lib.ModeKeys.TRAIN, self.config)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/estimator/estimator.py", line 790, in _call_model_fn
    model_fn_results = self._model_fn(features=features, **kwargs)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/contrib/timeseries/python/timeseries/head.py", line 228, in create_estimator_spec
    return self._train_ops(features)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/contrib/timeseries/python/timeseries/head.py", line 85, in _train_ops
    learning_rate=None)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/contrib/layers/python/layers/optimizers.py", line 241, in optimize_loss
    colocate_gradients_with_ops=colocate_gradients_with_ops)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/training/optimizer.py", line 458, in compute_gradients
    colocate_gradients_with_ops=colocate_gradients_with_ops)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py", line 610, in gradients
    lambda: grad_fn(op, *out_grads))
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py", line 376, in _MaybeCompile
    return grad_fn()  # Exit early
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py", line 610, in <lambda>
    lambda: grad_fn(op, *out_grads))
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/array_grad.py", line 589, in _PadGrad
    x_grad = array_ops.slice(grad, begin, sizes)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/array_ops.py", line 640, in slice
    return gen_array_ops._slice(input_, begin, size, name=name)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/gen_array_ops.py", line 4591, in _slice
    "Slice", input=input, begin=begin, size=size, name=name)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 510, in _apply_op_helper
    preferred_dtype=default_dtype)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1036, in internal_convert_to_tensor
    ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
  File "~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py", line 97, in _IndexedSlicesToTensor
    "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
  File "~/.pyenv/versions/3.6.4/lib/python3.6/warnings.py", line 99, in _showwarnmsg
    msg.file, msg.line)
~/.pyenv/versions/tf3/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py:97: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
```
I see discussion about this warning on SO:
https://stackoverflow.com/questions/35892412/tensorflow-dense-gradient-explanation/35893467
But the problem seems to happen in the estimator "train" method, not in any custom code I have written.

**Comment by drpngx** [comment]
You probably want to check the max index of your sparse tensor, or set the static shape to be sure.

This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
